conseil-api/src/test/scala/tech/cryptonomic/conseil/api/routes/platform/data/ethereum/EthereumDataOperationsTest.scala

```diff
@@ -94,6 +94,79 @@ class EthereumDataOperationsTest
+        dbHandler.run(Tables.Receipts ++= receipts).isReadyWithin(5.seconds) shouldBe true
+
+        whenReady(sut.fetchReceipts(Query.empty)) { result =>
+          result.value.size shouldBe 3
+        }
+      }
+
+      "return proper contracts, while fetching all contracts" in {
+        // given
+        dbHandler.run(Tables.Blocks ++= blocks).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.Transactions ++= transactions).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.Contracts ++= contracts).isReadyWithin(5.seconds) shouldBe true
+
+        whenReady(sut.fetchContracts(Query.empty)) { result =>
+          result.value.size shouldBe 3
+        }
+      }
+
+      "return proper tokens, while fetching all tokens" in {
+        // given
+        dbHandler.run(Tables.Blocks ++= blocks).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.Transactions ++= transactions).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.Tokens ++= tokens).isReadyWithin(5.seconds) shouldBe true
+
+        whenReady(sut.fetchTokens(Query.empty)) { result =>
+          result.value.size shouldBe 3
+        }
+      }
+
+      "return proper token transfers, while fetching all token transfers" in {
+        // given
+        dbHandler.run(Tables.Blocks ++= blocks).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.Transactions ++= transactions).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.Tokens ++= tokens).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.TokenTransfers ++= tokenTransfers).isReadyWithin(5.seconds) shouldBe true
+
+        whenReady(sut.fetchTokenTransfers(Query.empty)) { result =>
+          result.value.size shouldBe 3
+        }
+      }
+
+      "return proper number of accounts, while fetching all of accounts" in {
+        // given
+        dbHandler.run(Tables.Blocks ++= blocks).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.Transactions ++= transactions).isReadyWithin(5.seconds) shouldBe true
+
+        whenReady(sut.fetchAccounts(Query.empty)) { result =>
+          result.value.size shouldBe 2
+        }
+      }
+
+      "return proper account by address" in {
+        // given
+        dbHandler.run(Tables.Blocks ++= blocks).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.Transactions ++= transactions).isReadyWithin(5.seconds) shouldBe true
+
+        whenReady(sut.fetchAccountByAddress("to")) { result =>
+          result.value should (contain key "address" and contain value transaction1.to)
+          //TODO Fix test for 'value' field in AccountRow
+          //Issue with the current approach is that type inside case class is from 'scala' while slick is mapping data inside 'QueryResponse' to 'java'.
+          //Additionally BigDecimals are not easy comparable with 'scalatest'.
```

Comment:
Why exactly is not easy to compare BigDecimals? Maybe we can find a solution together?
Or is it just that we didn't yet decide what to use to describe the result, and this will be fixed then?

Reply:
I believe mostly because scaliest does not provide any support for comparing this types, but I remember that the fact that it was sometimes BigDecimal from scala package, sometimes from java package was painful in this test case, but since I have some time I can take a look at this with fresh mind and investigate that more.

Reply:
ow, the fact that it's changing between java/scala version is a nuisance, I have to admit...

What about extracting a comparable value from the big-decimal, like a asLong, considering that tests might be controlled to not overflow it?
Extraction could be done with pattern matching to handle both java and scala BD types...

Reply:
So I did investigated why we need to deal with `java.math.BigDecimal` instead of `scala.math.BigDecimal`.

Everything starts here: `tech.cryptonomic.conseil.common.util.DatabaseUtil.QueryBuilder#getMap`.
To access the data, while we are making generic query, we are using above `GetResult` implicit created only for this purpose. 

We are calling `positionedResult.nextObjectOption` in there. Looking into its implementation, you will see that this method is calling:
```
final def nextObjectOption()     = { val npos = pos + 1; val r = rs getObject     npos; val rr = (if(rs.wasNull) None else Some(r)); pos = npos; rr }
```

It works differently for `nextBigDecimalOption` tho:
```
final def nextBigDecimalOption() = { val npos = pos + 1; val r = rs getBigDecimal npos; val rr = (if(rs.wasNull) None else Some(BigDecimal(r))); pos = npos; rr }
```
So here implementation is wrapping `java.math.BigDecimal` into `scala.math.BigDecimal`, but we are not using this mechanism. Let's skip it for a while.

Gladly, Scala's implementation is just a wrapper, so we can call `.bigDecimal` and it should work like that:
```
result.value should (contain key "value" and contain value Some((transaction2.value + transaction3.value).bigDecimal))
```
... right? No... it's not going to work, because the scale for `BigDecimal` from SQL Query is `2`, while in our case it is `0`. You may ask, how the hell BigDecimals are comparable then? They are but not with `==`, which is used by default by `ScalaTest` while dealing with Collections/Maps. We should use `compareTo == 0` for that purpose.

Since we are not able to do that, my suggestion is to simply adjust the scale for this purpose to `2` and wrap that with comment, why we are doing that.